frontend/js/src/explore/fresh-releases/FreshReleases.tsx

```diff
@@ -217,29 +238,76 @@ export default function FreshReleases() {
         const releaseTags = Array.from(uniqueReleaseTagsSet);
         releaseTags.sort();
 
-        setReleases(cleanReleases);
-        setFilteredList(cleanReleases);
-        setAllFilters({
-          releaseTypes,
-          releaseTags,
-        });
-        setIsLoading(false);
+        return {
+          data: {
+            releases: cleanReleases,
+            releaseTypes,
+            releaseTags,
+          } as FreshReleasesData,
+          hasError: false,
+          errorMessage: "",
+        };
       } catch (error) {
-        toast.error(
-          <ToastMsg
-            title="Couldn't fetch fresh releases"
-            message={
-              typeof error === "object" ? error.message : error.toString()
-            }
-          />,
-          { toastId: "fetch-error" }
-        );
+        return {
+          data: {
+            releases: [],
+            releaseTypes: [],
+            releaseTags: [],
+          } as FreshReleasesData,
+          hasError: true,
+          errorMessage: error.message,
+        };
       }
-    };
-    // Call the async function defined above (useEffect can't return a Promise)
-    fetchReleases();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [pageType, range, showPastReleases, showFutureReleases, sort]);
+    },
+  });
+
+  const {
+    data: rawData = {
+      releases: [],
+      releaseTypes: [],
+      releaseTags: [],
+    } as FreshReleasesData,
+    hasError = false,
+    errorMessage = "",
+  } = loaderData || {};
+
+  const { releases, releaseTypes, releaseTags } = rawData;
+
+  React.useEffect(() => {
+    if (hasError) {
+      toast.error(
+        <ToastMsg
+          title="Couldn't fetch fresh releases"
+          message={errorMessage}
+        />,
+        { toastId: "fetch-error" }
+      );
+    }
+  }, [hasError, errorMessage]);
+
+  React.useEffect(() => {
+    if (releases.length > 0) {
+      setFilteredList(releases);
+    } else if (releases.length === 0 && filteredList.length > 0) {
+      setFilteredList([]);
+    }
+  }, [releases]);
+
+  React.useEffect(() => {
+    if (releaseTypes.length === 0 && releaseTags.length === 0) return;
+    setAllFilters({
+      releaseTypes,
+      releaseTags,
+    });
+  }, [releaseTypes, releaseTags]);
```

Comment:
Correct.
I believe I found the issue I was facing. When I was creating a map for `allFilters` and sending to the ReleaseFilters component, But the issue was that, on every re-render, this map was being created again and again. And since it's a `map`, it is compared by reference and not by value like other non-primitive type data types, causing the useEffect to trigger.